security/application.md

```diff
@@ -0,0 +1,412 @@
+# The thoughtbot Guide to Application Security
+
+## Threat modeling
+
+The task of identifying concrete attacks and understanding their relationship
+with the code is the core task of threat modeling. We can understand this from
+two perspectives:
+
+- identify what can go wrong, and
+- don't account for things that cannot go wrong.
+
+Identifying what can go wrong is what is most often [written about when
+discussing threat](https://www.owasp.org/index.php/Application_Threat_Modeling)
+modeling. There are [many
+techniques](https://insights.sei.cmu.edu/sei_blog/2018/12/threat-modeling-12-available-methods.html),
+but the summary is:
+
+1. Figure out what an attacker can do on your app. For a Web app, they might be
+   able to spoof HTTP headers, submit malicious data, or embed a Web page in an
+   `iframe`.
+2. Identify the weak points of the app. These will likely be places where you
+   are doing something non-standard, which the frameworks don't know how to
+   protect.
+3. Prioritize these. Take into account factors such as difficulty of attack,
+   likelihood of attack, ease of mitigating the attack, and severity of attack.
+
+Anything not in the list are things you cannot use as a reason to do something.
+Since the list is prioritized, you can use it to help it to prioritize tickets
+or split tickets.
+
+## Library updates
+
+The easiest line of defense we have as a developer is applying security fixes
+for our dependencies as they are released.
+
+On the flip perspective, when releasing a security fix for one of our projects,
+make it trivial to upgrade: don't include new features or unrelated bug fixes,
+if possible.
+
+There are a few ways to keep up with security fixes:
+
+- Any playform-specific tool, such as
+  [bundler-audit](https://github.com/rubysec/bundler-audit#readme).
+- The [#security](https://thoughtbot.slack.com/messages/security) channel.
```

Comment:
The guides are public, but this channel is not. Maybe call out that it's only for thoughtbot team members?